pypdf/_writer.py

```diff
@@ -369,6 +367,9 @@ def __exit__(
         """Write data to the fileobj."""
         if self.fileobj:
             self.write(self.fileobj)
+            close_attr = getattr(self.fileobj, "close", None)
+            if callable(close_attr):
+                self.fileobj.close()  # type: ignore[attr-defined]
```

Comment:
looking [@MasterOdin's  post](https://github.com/py-pdf/pypdf/issues/2905#issuecomment-2421498743) 

the stream closure should be done by the caller, not here, no ?

Reply:
IMHO it should be closed on exit to avoid leaking resources - unless I misunderstood the existing discussions.

Reply:
I agree that if a stream is opened by the PdfWriter(eg. if a path is provided)the stream should be closed but if it is the stream (opened before the context of the PdfWriter) it should be closed out of the 'with' section. As written it is always closed at the closure.